astroplan/core.py

```diff
@@ -35,13 +33,52 @@
 
 #__doctest_requires__ = {'*': ['scipy.integrate']}
 
+def _generate_24hr_grid(t0, start, end, N, for_deriv=False):
+    '''
+    Generate a nearly linearly spaced grid of time durations.
+
+    The midpoints of these grid points will span times from ``t0``+``start``
+    to ``t0``+``end``, including the end points, which is useful when taking
+    numerical derivatives.
+
+    Parameters
+    ----------
+    t0 : `~astropy.time.Time`
+        Time queried for, grid will be built from or up to this time.
+
+    start : float
+        Number of days before/after ``t0`` to start the grid.
+
+    end : float
+        Number of days before/after ``t0`` to end the grid.
+
+    N : int
+        Number of grid points to generate
+
+    for_deriv : bool
+        Generate time series for taking numerical derivative (modify
+        bounds)?
+
+    Returns
+    -------
+    `~astropy.time.Time`
+    '''
+
+    if for_deriv:
+        time_grid = np.concatenate([[start - 1/(N-1)],
+                                    np.linspace(start, end, N)[1:-1],
+                                    [end + 1/(N-1)]])*u.day
+    else:
+        time_grid = np.linspace(start, end, N)*u.day
+
+    return t0 + time_grid
 
 class Observer(object):
     """
     A container class for information about an observer's location and
     environment.
```

Comment:
We need to put a docstring in here for the class, but this could be done in a separate PR...